pkg/dependency/parser/nodejs/pnpm/parse.go

```diff
@@ -107,6 +141,97 @@ func (p *Parser) parse(lockVer float64, lockFile LockFile) ([]ftypes.Package, []
 	return pkgs, deps
 }
 
+func (p *Parser) parseV9(lockFile LockFile) ([]ftypes.Package, []ftypes.Dependency) {
+	lockVer := 9.0
+	resolvedPkgs := make(map[string]ftypes.Package)
+	resolvedDeps := make(map[string]ftypes.Dependency)
+
+	// Check all snapshots and save with resolved versions
+	resolvedSnapshots := make(map[string][]string)
+	for depPath, snapshot := range lockFile.Snapshots {
+		name, version := p.parseDepPath(depPath, lockVer)
+
+		var dependsOn []string
+		for depName, depVer := range lo.Assign(snapshot.OptionalDependencies, snapshot.Dependencies) {
+			depVer = p.trimPeerDeps(depVer, lockVer) // pnpm has already separated dep name. therefore, we only need to separate peer deps.
+			depVer = p.parseVersion(depPath, depVer, lockVer)
+			id := packageID(depName, depVer)
+			if _, ok := lockFile.Packages[id]; ok {
+				dependsOn = append(dependsOn, id)
+			}
+		}
+		if dependsOn != nil {
```

Comment:
nit: https://go.dev/play/p/DtIqxxy-mLq
```suggestion
		if len(dependsOn) != 0 {
```

Reply:
BTW: I want golangci-lint to enforce it.

Reply:
fixed in https://github.com/aquasecurity/trivy/pull/6617/commits/535b1e22b44c990578f3f04ea88b235abf0b28be